docs/src/fields.md

```diff
@@ -0,0 +1,331 @@
+```
+
+We can also use arrays,
+
+```@setup setting
+using Random
+Random.seed!(123)
+```
+
+```jldoctest setting
+random_stuff = rand(size(c)...)
+set!(c, random_stuff)
+
+# output
+4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
+├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: ZeroFlux, top: ZeroFlux, immersed: ZeroFlux
+└── data: 10×10×10 OffsetArray(::Array{Float64, 3}, -2:7, -2:7, -2:7) with eltype Float64 with indices -2:7×-2:7×-2:7
+    └── max=0.991405, min=0.0303789, mean=0.520014
+```
+
+and even functions,
+
+```jldoctest setting
+fun_stuff(x, y, z) = 2x
+
+set!(c, fun_stuff)
+
+# output
+4×4×4 Field{Center, Center, Center} on RectilinearGrid on CPU
+├── grid: 4×4×4 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 3×3×3 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: ZeroFlux, top: ZeroFlux, immersed: ZeroFlux
+└── data: 10×10×10 OffsetArray(::Array{Float64, 3}, -2:7, -2:7, -2:7) with eltype Float64 with indices -2:7×-2:7×-2:7
+    └── max=7.0, min=1.0, mean=4.0
+```
+
+For `Field`s on three-dimensional grids, the functions must have arguments `x, y, z` for `RectilinearGrid`, or `λ, φ, z` for `LatitudeLongitudeGrid` and `OrthogonalSphericalShellGrid`.
+But for `Field`s on one- and two-dimensional grids, only the non-`Flat` directions are included.
+For example
+
+```jldoctest setting
+one_d_grid = RectilinearGrid(size=7, x=(0, 7), topology=(Periodic, Flat, Flat))
+one_d_c = CenterField(one_d_grid)
+more_fun_stuff(x) = 3x
+set!(one_d_c, more_fun_stuff)
+
+# output
+7×1×1 Field{Center, Center, Center} on RectilinearGrid on CPU
+├── grid: 7×1×1 RectilinearGrid{Float64, Periodic, Flat, Flat} on CPU with 3×0×0 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Periodic, east: Periodic, south: Nothing, north: Nothing, bottom: Nothing, top: Nothing, immersed: ZeroFlux
+└── data: 13×1×1 OffsetArray(::Array{Float64, 3}, -2:10, 1:1, 1:1) with eltype Float64 with indices -2:10×1:1×1:1
+    └── max=19.5, min=1.5, mean=10.5
+```
+
+Now, since
+
+```jldoctest setting
+xnodes(c)
+
+# output
+4-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, -2:7), 1:4) with eltype Float64:
+ 0.5
+ 1.5
+ 2.5
+ 3.5
+```
+
+we find that
+
+```jldoctest setting
+c[1:4, 1, 1]
+
+# ouptut
+4-element Vector{Float64}:
+ 1.0
+ 3.0
+ 5.0
+ 7.0
+```
+
+In other words, `fun_stuff` is evaluated using the coordinates at the center of each cell for a given `Field`.
+One consequence is that the result is different for a `Field` located at `x` faces,
+
+```jldoctest setting
+u = XFaceField(grid)
+set!(u, fun_stuff)
+u[1:4, 1, 1]
+
+# output
+4-element Vector{Float64}:
+ 0.0
+ 2.0
+ 4.0
+ 6.0
+```
```

Comment:
I felt that this section should come _before_ the bit: "For `Field`s on three-dimensional grids, the function...."

Reply:
Let me think about that. Also the next section is going to go into halo regions and this leads naturally into that. Maybe.

Reply:
I see what you're saying

Reply:
If we keep it as is, then perhaps half a phrase to ensure the people know that now we are connecting to the example one up... E.g. "Going back to the case where we set `c` with 2x, now, since...."